recipe/migrations/hdf51120.yaml

```diff
@@ -0,0 +1,7 @@
+__migrator:
+  build_number: 1
+  kind: version
+  migration_number: 1
+hdf5:
+- 1.12.0
```

Comment:
```suggestion
- 1.12.0
- 1.10.5
```

Reply:
Looks fine to me

Reply:
What about `1.10.6` that is currently in migration?

Reply:
Ugh. We should wait for that. I forgot.

Reply:
Well, can we stop that migration?

Reply:
We can pause it by adding the `pause` keyword to the migrator YAML metadata. It might be `paused` actually, but why? Do we want to start the 1.12.0 migration soon?

Reply:
Let's ask @djhoese!

Reply:
:fearful: I'm primarily a python programmer and h5py hasn't had a release that supports HDF 1.12 yet so I see no rush to do the migration. That said, given that HDF5 1.12 has large API changes that may require additional flags to build against it, it is probably best to find those issues sooner rather than later.

From the release notes for HDF5:

```
Compatibility with Previous Releases

This release required changes to existing APIs. Applications that were created with earlier HDF5 releases may not compile with 1.12 by default. The API Compatibility Macros in HDF5 allow users to work around this issue. Users can specify a compatibility macro mapping for the version of HDF5 that an application was built with. For example, a 1.10 application can be built with 1.12 using either an application or library mapping as follows:

    To compile an application built with a version of HDF5 that includes deprecated symbols (the default), specify: -DH5_USE_110_API (autotools) or –DH5_USE_110_API:BOOL=ON (CMake)
    To build an HDF5 library with the 1.10 APIs specify --with-default-api-version=v110 (Autotools) or -DDEFAULT_API_VERSION:STRING=v110 (CMake).

However, users will not be able to take advantage of some of the new features in 1.12 if using these compatibility mappings.

Also be aware that the CMake minimum version is now 3.12 and CMake 3.15 is recommended for the best build experience. Windows requires a minimum of CMake 3.14 and CMake 3.15 is required for VS 2019 support.

```